api/v1alpha1/nodepool_types.go

```diff
@@ -317,6 +317,49 @@ type NodePoolPlatform struct {
 	Agent *AgentNodePoolPlatform `json:"agent,omitempty"`
 
 	Azure *AzureNodePoolPlatform `json:"azure,omitempty"`
+
+	// PowerVS specifies the configuration used when using IBMCloud PowerVS platform.
+	//
+	// +optional
+	PowerVS *PowerVSNodePoolPlatform `json:"powervs,omitempty"`
+}
+
+// PowerVSNodePoolPlatform specifies the configuration of a NodePool when operating
+// on IBMCloud PowerVS platform.
+type PowerVSNodePoolPlatform struct {
+	// SysType used to host the instance(e.g: s922, e980, e880)
+	// +optional
+	SysType string `json:"sysType,omitempty"`
+
+	// ProcType (dedicated, shared, capped)
+	// +kubebuilder:default=shared
+	// +kubebuilder:validation:Enum=dedicated;shared;capped
+	// +optional
+	ProcType string `json:"procType,omitempty"`
+
+	// Processors specifies the number of processors allocated
+	// +optional
+	Processors string `json:"processors,omitempty"`
+
+	// Memory specifies the amount of memory specified in GBs
+	// +optional
+	Memory string `json:"memory,omitempty"`
+
+	// Image used for deploying the nodes
+	// +optional
```

Comment:
why is this optional, what happen if none is set?

Reply:
if none is set then we import the image from the release image, will enhance the documentation.